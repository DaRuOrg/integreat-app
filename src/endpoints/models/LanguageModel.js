class LanguageModel {
  constructor (code, name) {
    this._name = code
    this._code = name
  }

  get code () {
    return this._name
  }

  get name () {
    return this._code
  }
}

export default LanguageModel
