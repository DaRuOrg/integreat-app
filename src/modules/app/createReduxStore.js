// @flow

import type { Store } from 'redux'
import { applyMiddleware, combineReducers, compose, createStore } from 'redux'
import { createLogger } from 'redux-logger'

import { languageReducer, uiDirectionReducer } from '../i18n/reducers'
import toggleDarkModeReducer from '../theme/reducers'
import Payload from '../endpoint/Payload'

export type ActionType<T> = { type: string, payload: Payload<T> }

// todo: Change type to correct State type,
// https://blog.callstack.io/type-checking-react-and-redux-thunk-with-flow-part-2-206ce5f6e705
const createReduxStore = (initialState: {} = {}): Store<any, any> => {
  /**
   * The middlewares of this app, add additional middlewares here
   */
  const middlewares = []

  if (__DEV__) {
    middlewares.push(createLogger()) // Logs all state changes in console
  }

  const rootReducer = combineReducers({
    uiDirection: uiDirectionReducer,
    language: languageReducer,
    darkMode: toggleDarkModeReducer
  })

  const enhancers = compose(applyMiddleware(...middlewares))

  return createStore(rootReducer, initialState, enhancers)
}

export default createReduxStore
