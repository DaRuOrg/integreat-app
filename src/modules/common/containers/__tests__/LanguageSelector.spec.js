import React from 'react'
import { mount, shallow } from 'enzyme'
import { Provider } from 'react-redux'

import ConnectedLanguageSelector, { LanguageSelector } from '../LanguageSelector'
import createReduxStore from '../../../app/createReduxStore'
import createHistory from '../../../app/createHistory'
import LanguageModel from '../../../endpoint/models/LanguageModel'

describe('LanguageSelector', () => {
  const location = 'augsburg'
  const language = 'de'
  const languages = [
    new LanguageModel('de', 'Deutsch'),
    new LanguageModel('en', 'English'),
    new LanguageModel('ar', 'Arabic')
  ]

  const languageChangeUrls = {
    en: 'test/url/en',
    de: 'test/url/de',
    ar: 'test/url/ar'
  }

  test('should match snapshot', () => {
    const mockCloseDropDownCallback = jest.fn()

    const wrapper = shallow(
      <LanguageSelector location={location}
                        languageChangeUrls={languageChangeUrls}
                        closeDropDownCallback={mockCloseDropDownCallback}
                        language={language}
                        languages={languages} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  describe('connect', () => {
    test('should have correct props', () => {
      const mockCloseDropDownCallback = jest.fn()

      const store = createReduxStore(createHistory, {
        router: {params: {location: 'augsburg', language: 'de'}},
        languageChangeUrls: languageChangeUrls
      })

      const tree = mount(
        <Provider store={store}>
          <ConnectedLanguageSelector closeDropDownCallback={mockCloseDropDownCallback} />
        </Provider>
      )

      const languageSelector = tree.find(LanguageSelector)

      expect(languageSelector.props()).toEqual({
        closeDropDownCallback: mockCloseDropDownCallback,
        languages: languages,
        language: language,
        location: location,
        languageChangeUrls: languageChangeUrls,
        // props of withFetcher
        dispatch: expect.any(Function),
        languagesPayload: expect.any(Object),
        requestAction: expect.any(Function),
        urlParams: expect.any(Object)
      })
    })
  })
})
