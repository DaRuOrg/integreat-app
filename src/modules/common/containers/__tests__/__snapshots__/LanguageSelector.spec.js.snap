// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector should match snapshot 1`] = `
<div
  className="languageSelector"
>
  <span
    className="element elementActive"
  >
    Deutsch
  </span>
  <Connect(Link)
    className="element"
    href="test/url/en"
    key="en"
    onClick={[Function]}
  >
    English
  </Connect(Link)>
  <Connect(Link)
    className="element"
    href="test/url/ar"
    key="ar"
    onClick={[Function]}
  >
    Arabic
  </Connect(Link)>
</div>
`;
